### ISSUE TITLE
num_supported_devices in connect() adds test time

### ISSUE THREAD

**Issue by wallacbe** [square/pylink#54]
num_supported_devices() call inconnect adds > 100 ms test time each call.  If we are not using lock/unlock can we add flag argument to skip?

https://github.com/square/pylink/blob/master/pylink/jlink.py#L933

**Comment by hkpeprah** [comment]
I would rather not add a flag for this, but I think making `_device` be a property on doing it when needed makes sense, and would produce the same result.  E.g.

```
def connect(...):
    <snipped>
    self._chip_name = chip_name

def disconnect(...):
    self._device = None

@property
def device(self):
    if self._device:
        return self._device
    <do lookup>
    return instance
```

If you want to go ahead and submit a patch, I'd be more than happy to merge it.

**Comment by hkpeprah** [comment]
This has been resolved with #74 .
